spec/lib/appsignal/config_spec.rb

```diff
@@ -347,6 +347,142 @@ def on_load
         end
       end
     end
+
+    describe ":revision" do
+      let(:revision_file_path) { File.join(tmp_dir, "REVISION") }
+      let(:config) { silence { build_config(:env => :none, :root_path => tmp_dir) } }
+
+      context "when REVISION file exists and is readable" do
+        before do
+          File.write(revision_file_path, "abc123\n")
+        end
+
+        it "sets the revision from the file" do
+          expect(config[:revision]).to eq("abc123")
+        end
+
+        it "sets the revision as loaded through the system" do
+          expect(config.system_config).to include(:revision => "abc123")
+        end
+
+        it "logs successful file reading" do
+          logs = capture_logs { build_config(:env => :none, :root_path => tmp_dir) }
+          expect(logs).to contains_log(:debug,
+            "REVISION file found and read successfully at: #{revision_file_path}")
+        end
```

Comment:
```suggestion
      context "with REVISION file content" do
        before do
          File.write(revision_file_path, "abc123\n")
        end

        it "sets the revision from the file" do
          expect(config[:revision]).to eq("abc123")
        end

        it "sets the revision as loaded through the system" do
          expect(config.system_config).to include(:revision => "abc123")
        end

        it "logs successful file reading" do
          logs = capture_logs { build_config(:env => :none, :root_path => tmp_dir) }
          expect(logs).to contains_log(:debug,
            "REVISION file found and read successfully at: #{revision_file_path}")
        end
      end
```

Let's move this to a block so that the before block isn't run before the before blocks from the other context blocks, which would mean it writes the revision file multiple times in one test.